classes/output/mobile.php

```diff
@@ -61,10 +61,28 @@ public static function forum_discussions_view($args) {
         $allgroups              = groups_get_all_groups($cm->course, 0, $cm->groupingid);
         $allowedgroups          = groups_get_activity_allowed_groups($cm, $USER->id);
 
-        if (count($allowedgroups) && (int) $cm->groupmode > 0) {
-            $showgroupsections = true;
+        // Check what groups the user is allowed to see/post to.
+        if ((int) $cm->groupmode > 0) {
```

Comment:
If you moved this if-else block out into a function you could simplify it with a guarded clause.